src/export/dashboard.ts

```diff
@@ -0,0 +1,58 @@
+// (C) 2019 GoodData Corporation
+import { DashboardExport } from "@gooddata/typings";
+import { ApiResponse, ApiResponseError, XhrModule } from "../xhr";
+import { IExportResponse } from "../interfaces";
+import { handleHeadPolling, IPollingOptions } from "../util";
+import { sanitizeDateFilters, isDataExported } from "./util";
+
+interface IExportDashboardPayload {
+    dashboardExport: {
+        dashboardUri: string;
+        filters?: DashboardExport.FilterContextItem[];
+    };
+}
+
+export class DashboardModule {
+    constructor(private xhr: XhrModule) {}
+
+    public exportDashboardToPdf(
+        projectId: string,
+        dashboardUri: string,
+        filters: DashboardExport.FilterContextItem[] = [],
+        pollingOptions: IPollingOptions = {},
+    ): Promise<IExportResponse> {
+        const convertedDateFilters = sanitizeDateFilters(filters);
+        const payload = this.getExportDashboardPayload(dashboardUri, convertedDateFilters);
+
+        return this.xhr
+            .post(`/gdc/internal/projects/${projectId}/exportDashboard`, { body: payload })
+            .then(this.getPdfUri)
+            .then(this.pollPdfFile.bind(this, pollingOptions))
```

Comment:
please use async-await approach with try-catch instead of chaining promises it will be more readable and without `binding` arguments...

also consider to not use too much granular methods as `handleError` :)

Reply:
Fixed after using `async/await`